_posts/2022-10-09-decentralization-of-zk-rollups.md

```diff
@@ -0,0 +1,192 @@
+author: Ibrahim Yusufali
+title: Decentralization of ZK Rollups
+excerpt: "Zero knowledge rollups were used in the past couple of years to solve the scalability constraints in L1 monolithic chains. A lot of new projects have emerged in the space since then, each of them ambitious and hoping to be the “ultimate” solution. However, as it is with every rapidly growing technology there are certain aspects that may be overlooked, in order to facilitate fast execution and entry to the market. We will be doing a deep dive into what we believe is one of those aspects: “decentralization”."
+---
+
+_Many thanks to Preston Evans, Shashank Agrawal, Ventali Tan, Daniel Lubarov, Gautam Botrel and James Stearn for feedback._
+
+## Outline
+
+1. Introduction
+2. Upgrade Keys
+3. Why Proof of Stake (PoS) Might Not Work
+4. Proof of Efficiency
+5. Proof Generation Outsourcing + MEV Auction
+6. Our Proposals
+7. Conclusion
+
+
+## Introduction
+
+Zero knowledge rollups were used in the past couple of years to solve the scalability constraints in L1 monolithic chains. A lot of new projects have emerged in the space since then, each of them ambitious and hoping to be the “ultimate” solution. However, as it is with every rapidly growing technology there are certain aspects that may be overlooked, in order to facilitate fast execution and entry to the market. We will be doing a deep dive into what we believe is one of those aspects: “decentralization”. 
+
+All current zk rollups are centralized. Some do not even have a projected path to decentralization. This is not a good situation to be in. Some features in a truly decentralized zk-rollup which we hope to achieve:
+
+
+
+1. Safety / Soundness. Ability to resist a malicious party from taking control over the network, and dictating the present and future state of the network. 
+2. Censorship resistance. Everyone should be able to have access to the network and interact with other participants. The network should not be able to deny this right to any individual. It should also be possible for users to withdraw their funds back to L1 or force execute transactions in L2 by paying gas in L1, even if the network is compromised, or the operators are limiting user activities on the network. 
+3. Low node / network participant requirements. Individuals should be able to build and run their own node, prover or sequencer without an unreasonable amount of difficulty. Expecting state of the art or highly specialized equipment and high compute resources should be avoided. 
+4. Economically aligned incentives. Participants should be rewarded equally for upkeep of the network, in order to encourage new participants to enter. 
+5. Liveness. Honest messages should always be included and available to block producers. Chances of halting or lapses in network availability  should be mitigated. 
+
+
+## Upgrade Keys
+
+Upgrade keys are public keys that have been whitelisted by a rollup’s L1 smart contract, and they have two main benefits:
```

Comment:
Are "Upgrade keys" specifically referring to keys or addresses that have the power to change the implementation of the entire contract? I think the scope and power of such keys in discussion should be defined here in more detail (instead of just stating "benefits"). There are also many different types of upgradability: beacon-based upgradability, delayed upgradability, multi-facet upgradability (EIP2535), and some others. Each has different level of security exposure and risk management mechanism. I think specifically the multi-facet version should be given a lot more discussions. Some aspects of the contract (such as how a ZKP proof is verified) could be left immutable (or at least unchanged) during an upgrade, whereas some others do not. This allows different components enforcing different security requirements and precautions (such as delay) to execute an upgrade

Reply:
This is a valid point, thank you for raising it. I have honestly not gone in-depth with upgradability, and am of the opinion that this section does not really fit well with the overall article anyways, and is not related to the main message. Is it best to remove this section of Upgrade Keys, for the reason that to do it service, we need to extend it, but then this might give it more attention than we want it to?

Reply:
The issue is all discussions below in this section assume "upgrade keys" are one or a few keys that have all the power and control over the L2, and argue upgrade keys overall are problematic. This is faulty logic

Reply:
Understood. So I will remove this section then.